File: content/python/templates/gcd.py
# Greatest Common Divisor (GCD)

def gcd(a, b):
    """Calculate GCD using Euclidean algorithm."""
    while b:
        a, b = b, a % b
    return a

# Recursive version
def gcd_recursive(a, b):
    """Calculate GCD recursively."""
    if b == 0:
        return a
    return gcd_recursive(b, a % b)

# Test
a, b = 48, 18
print(f"GCD({a}, {b}) = {gcd(a, b)}")

# Using math module
import math
print(f"GCD (math): {math.gcd(a, b)}")

File: content/java/templates/06_Algorithms/gcd.java
// Greatest Common Divisor (GCD)

public class GCD {
    // Euclidean algorithm
    public static int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    
    // Recursive version
    public static int gcdRecursive(int a, int b) {
        if (b == 0) return a;
        return gcdRecursive(b, a % b);
    }
    
    public static void main(String[] args) {
        int a = 48, b = 18;
        
        System.out.println("GCD(" + a + ", " + b + ") = " + gcd(a, b));
        System.out.println("GCD (recursive) = " + gcdRecursive(a, b));
    }
}
